storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -669,4 +705,8 @@ public static String remoteTransactionIndexFileName(RemoteLogSegmentMetadata rem
         return generateFileNamePrefixForIndex(remoteLogSegmentMetadata) + LogFileUtils.TXN_INDEX_FILE_SUFFIX;
     }
 
+    public static int estimatedEntrySize(Entry entry) {
```

Comment:
I have updated it, but the embarrassing thing is: even if long is returned here, the weighter implementation in Caffeine cache still needs to convert long to int, because the return type of weigher is int. @divijvaidya